h/templates/accounts/base.html.jinja2

```diff
@@ -0,0 +1,33 @@
+{% extends "h:templates/layouts/base.html.jinja2" %}
+
+{% block header %}
+  {% include "h:templates/includes/logo-header.html.jinja2" %}
+{% endblock %}
+
+{% block content %}
+  <div class="form-container content">
+    {% include "h:templates/includes/flash-messages.html.jinja2" %}
+
+    <h1 class="form-header">
+      {% block page_title %}{% endblock %}
+    </h1>
+
+    <div class="form-description">
+      {% block form_message %}{% endblock %}
+    </div>
+
+    {% if form %}<p>{{ form }}</p>{% endif %}
+
+    <footer class="form-footer">
+      {% block form_footer %}{% endblock %}
+      {% if signup_message %}
+        {{ signup_message }}
+        <a class="link--footer" href="{{ request.route_path('signup') }}">Sign up</a>
+      {% endif %}
+      {% if login_message %}
+        {{ login_message }}
+        <a class="link--footer" href="{{ request.route_path('login') }}">Log in</a>
+      {% endif %}
+    </footer>
+  </div>
+{% endblock content %}
```

Comment:
## Anatomy of a sign up / log in page

All of the various sign up and log in-related pages follow the same design, with a fixed set of components always in the same layout. Each page only uses some of the components. The components never all appear together on a single page.

Here's a fake page demonstrating all of the components:

![Screenshot from 2020-03-20 19-01-31](https://user-images.githubusercontent.com/22498/77197635-73b85b80-6add-11ea-8279-3b59e115583f.png)

The components are:

1. The h logo.

2. Flash messages. Handled by the `{% include "h:templates/includes/flash-messages.html.jinja2" %}` in this base template. This means that all of these pages now have flash messages support. In practice flash messages are only ever written before loading the login page though.

3. The page title. Also used as the tab/window title.

4. The "form message". To get one of these a page's template would do `{% block form_message %}This is the form message{% endblock %}`.

5. The form. Appears if there's a `form` variable in the template context. Put there by the Python code. Some pages have a form, some don't.

6. The footer with sign up and log in links. In practice either the sign up or log in link always appears, never both. The text below the link varies from page to page. To have a sign up link a page's template would do `{% set signup_message = "Don't have a Hypothesis account?" %}`. Similarly to have a log in link it would set a `login_message` variable.